support-figma/extended-layout-plugin/src/ui.html

```diff
@@ -759,63 +798,141 @@
+    const td = document.createElement("td");
+    let extras = stringRes["extras"];
+    if (!extras) {
+      extras = {"charlimit": "NONE", "description": ""};
+      stringRes["extras"] = extras;
+    }
+    const textBlock = document.createElement("div");
+    textBlock.textContent = JSON.stringify(extras);
+    td.appendChild(textBlock);
+    if (stringRes["textNodes"] != null) {
+      const iconButton = document.createElement("div");
+      iconButton.className = "icon-button";
+      const svgElement = document.createElementNS("http://www.w3.org/2000/svg", "svg");
+      const useElement = document.createElementNS("http://www.w3.org/2000/svg", "use");
+      useElement.setAttributeNS("http://www.w3.org/1999/xlink", "xlink:href", "#icon-edit");
+      svgElement.appendChild(useElement);
+      iconButton.appendChild(svgElement);
+      td.appendChild(iconButton);
+
+      const form = document.createElement("form");
+      form.style.display="none";
+
+      const charlimitInput = document.createElement("input");
+      charlimitInput.type = "number";
+      charlimitInput.min = "0";
+      charlimitInput.max = "200";
+      charlimitInput.step = "5";
+      charlimitInput.name = "charlimit";
+      charlimitInput.placeholder = "Character Limit";
+      charlimitInput.value = extras["charlimit"] == "NONE" ? 0 : extras["charlimit"];
+
+      const descriptionInput = document.createElement("input");
+      descriptionInput.type = "text";
+      descriptionInput.name = "description";
+      descriptionInput.placeholder = "Enter description";
+      if (extras["description"] != null) {
+        descriptionInput.value = extras["description"];
+      }
+
+      // Create a submit button (optional)
+      const submitButton = document.createElement("button");
+      submitButton.type = "submit";
+      submitButton.textContent = "Submit";
+
+      // Append the input fields and button to the form
+      form.appendChild(charlimitInput);
+      form.appendChild(document.createElement("br"));
+      form.appendChild(descriptionInput);
+      form.appendChild(document.createElement("br"));
+      form.appendChild(submitButton);
+      td.appendChild(form);
+
+      form.addEventListener("submit", (event) => {
+        event.preventDefault(); // Prevent default form submission
+        const charlimit = charlimitInput.value;
+        const description = descriptionInput.value;
+        extras["charlimit"] = charlimit == 0 ? "NONE" : charlimit;
+        extras["description"] = description;
+        textBlock.textContent = JSON.stringify(extras);
+        form.style.display = "none";
+        iconButton.style.display="flex";
+        parent.postMessage({ pluginMessage: {
+            msg: 'update-localization-data',
+            item: stringRes
+          }}, '*');
+      });
+      iconButton.addEventListener("click", () => {
+        form.style.display = "block";
+        iconButton.style.display = "none";
+      });
+    }
+    return td;
+  }
+
```

Comment:
I think this is ok for now, but it would be really nice to have some design support for this UI. @iamralpht, Babak could you help get some? I think the "Extras" column is a little clunky with the way it displays values as JSON and lets you edit the values in a text field. I personally probably would just have a char limit column with a text field.

Reply:
Filed ticket #1295 to track this.

I updated the description to align with what currently google uses: `description [CHAR_LIMIT=number|NONE]`. Keep it in one column but it now only shows the input boxes.
<img width="327" alt="Screenshot 2024-06-26 at 10 32 48" src="https://github.com/google/automotive-design-compose/assets/165221276/9ee3c897-d441-4b26-a2fe-0ad93a69ad9f">